core/src/types.ts

```diff
@@ -266,3 +266,53 @@ export declare interface DebtAuctionEnriched extends DebtAuctionActive {
 }
 
 export declare interface DebtAuctionTransaction extends DebtAuctionEnriched, CompensationAuctionTransactionFees {}
+
+type CollateralType = CollateralConfig['title'];
+export declare interface VaultCollateralType {
+    collateralType: CollateralType;
+}
+
+export declare interface LiquidationLimit {
+    maximumAmountAuctionedCollateral: BigNumber;
+    currentlyAuctionedCollateral: BigNumber;
+}
+
+export declare interface VaultCollateralParameters extends LiquidationLimit {
+    stabilityFeeRate: BigNumber;
+    maxDebtPerCollateralUnit: BigNumber;
+}
+
+export declare interface VaultBase extends VaultCollateralType {
+    id: number;
+    address: string;
+}
+
+export declare interface VaultAmount {
+    initialDebtDai: BigNumber;
+    collateralAmount: BigNumber;
+}
+
+export declare interface VaultTransactionFees {
+    transactionFeeLiquidationEth: BigNumber;
+    transactionFeeLiquidationDai: BigNumber;
+}
+
+export declare interface VaultBaseInfo extends VaultBase, VaultAmount {}
+export declare interface OraclePrices {
+    currentUnitPrice: BigNumber;
+    nextUnitPrice: BigNumber;
+    nextPriceChange: Date;
+}
+
+export declare interface Vault extends VaultBaseInfo, VaultTransactionFees, OraclePrices {
+    liquidationRatio: number;
+    collateralizationRatio: number;
+    proximityToLiquidation: number;
+    state: 'liquidated' | 'liquidatable' | 'not-liquidateable';
+    incentiveRelativeDai: BigNumber;
+    incentiveConstantDai: BigNumber;
+    incentiveCombinedDai: BigNumber;
+    grossProfitDai: BigNumber;
+    netProfitDai: BigNumber;
+    debtDai: BigNumber;
```

Comment:
The type that assembles everything together and mostly computes (profits, fees, etc) historically called `*Transaction` in our codebase. You can also adhere to that and create a separate type `VaultTransaction`.

To help you will separation: `Vault` type will be used by twitter bot to announce that there is a vault in the `liquidatable` state, `VaultTransaction` type will be used by the keeper/frontend to participate in it